internal/backend/db/dbgorm/projects.go

```diff
@@ -28,6 +30,44 @@ func (db *gormdb) CreateProject(ctx context.Context, p sdktypes.Project) error {
 	return translateError(db.createProject(ctx, project))
 }
 
+func (db *gormdb) deleteProject(ctx context.Context, projectID string) error {
+	p := scheme.Project{ProjectID: projectID}
+	return db.db.WithContext(ctx).Delete(&p).Error
+}
+
+func (db *gormdb) deleteProjectAndDependents(ctx context.Context, projectID string) error {
+	return db.transaction(ctx, func(tx *tx) error {
+		depsEnvs, err := db.getProjectDeployments(ctx, projectID)
+		if err != nil {
+			return err
+		}
+		var deplIDs, envIDs []string
+		for _, de := range depsEnvs {
+			if de.State != int32(sdktypes.DeploymentStateInactive.ToProto()) {
+				return fmt.Errorf("%w: project <%s>: cannot delete non-inactive deployment <%s> in state <%s>",
+					sdkerrors.ErrFailedPrecondition, projectID, de.DeploymentID,
+					deploymentsv1.DeploymentState(de.State).String())
+			}
+			deplIDs = append(deplIDs, de.DeploymentID)
+			envIDs = append(envIDs, de.EnvID)
+		}
+
+		if err = db.deleteDeploymentsAndDependents(ctx, deplIDs); err != nil {
+			return err
+		}
+
+		if err = db.deleteEnvs(ctx, envIDs); err != nil {
```

Comment:
this assumes all envs has a deployment, 
but if you just create an env, and don't deploy anything, you won't get this envID in the list and won't delete it
so you have to check for all project envs directly, not only through deployments